File: dcp-solutions/2024-12-21/solution.go
```go
func subarraySum(nums []int, k: int) []int {
    sumMap := make(map[int]int)
    sumMap[0] = 1
    
    total := 0
    result := []int{}
    
    for _, num := range nums {
        total += num
        
        if value, ok := sumMap[k-total]; ok {
            result = append(result, k-total+1)
        }
        
        sumMap[total]++
    }
    
    return result
}
```

File: dcp-solutions/2024-12-21/solution.js
```javascript
function subarraySum(nums, k) {
    let sumMap = new Map();
    sumMap.set(0, -1);
    let currentSum = 0;
    
    for (let i = 0; i < nums.length; i++) {
        currentSum += nums[i];
        
        if (sumMap.has(currentSum - k)) {
            return [nums[sumMap.get(currentSum - k) + 1], currentSum, i].slice(1);
        }
        
        sumMap.set(currentSum, i);
    }
    
    return [];
}
```